js/btcmarkets.js

```diff
@@ -243,26 +262,25 @@ module.exports = class btcmarkets extends Exchange {
         }
         const fee = this.safeFloat (transaction, 'fee');
         const status = this.parseTransactionStatus (this.safeString (transaction, 'status'));
-        const ccy = this.safeString (transaction, 'currency');
+        const ccy = this.safeString (transaction, 'assetName');
         const code = this.safeCurrencyCode (ccy);
-        // for some currencies the exchange requires the amount to be scaled, like BTC, ETH, BCH
-        // values in other currencies may have to be treated "as is", without scaling, like XRP
-        // https://github.com/ccxt/ccxt/issues/7413
-        let amount = this.safeValue (transaction, 'amount');
+        // todo: this logic is duplicated below
+        let amount = this.safeFloat (transaction, 'amount');
         if (amount !== undefined) {
+            amount = amount * 1e-8;
```

Comment:
This line doesn't look totally right to me, I mean we are dividing the amount by 100000000 and then a few lines below in the `else` case the amount is divided by 100000000 again.
Since the amount is always a float in this PR, we shouldn't check the string type in line 271.

Also, we should prefer division by 100000000 over multiplication by 1e-8, because in JS that helps avoid some fp-errors like `159.000000000000000001`.

@npomfret lmk your thoughts